src/client/cli/cmd/launch.cpp

```diff
@@ -92,26 +89,27 @@ auto net_digest(const QString& options)
             else if (key == "mac")
                 net.set_mac_address(checked_mac(val.toStdString()));
             else
-                throw NetworkDefinitionException{fmt::format("Bad network field: {}", key)};
+                throw mp::ValidationException{fmt::format("Bad network field: {}", key)};
         }
 
         // Interpret as "name" the argument when there are no ',' and no '='.
         else if (key_value_split.size() == 1 && split.size() == 1)
             net.set_id(key_value_split[0].toStdString());
 
         else
-            throw NetworkDefinitionException{fmt::format("Bad network field definition: {}", key_value_pair)};
+            throw mp::ValidationException{fmt::format("Bad network field definition: {}", key_value_pair)};
     }
 
     if (net.id().empty())
-        throw NetworkDefinitionException{fmt::format("Bad network definition, need at least a 'name' field")};
+        throw mp::ValidationException{fmt::format("Bad network definition, need at least a 'name' field")};
 
     return net;
 }
 } // namespace
 
 mp::ReturnCode cmd::Launch::run(mp::ArgParser* parser)
 {
+    arg_parser = parser;
```

Comment:
Why this? Better just pass it (or the timeout directly) to `request_launch` and prevent mismatches.